model/resource/cloud_provider/aws/ecs.yaml

```diff
@@ -28,15 +28,23 @@ groups:
       - id: task.arn
         type: string
         brief: >
-          The ARN of an [ECS task definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html).
-        examples: ['arn:aws:ecs:us-west-1:123456789123:task/10838bed-421f-43ef-870a-f43feacbbb5b']
+          The ARN of a running [ECS task](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids).
+        examples: [
+          'arn:aws:ecs:us-west-1:123456789123:task/10838bed-421f-43ef-870a-f43feacbbb5b',
+          'arn:aws:ecs:us-west-1:123456789123:task/my-cluster/task-id/23ebb8ac-c18f-46c6-8bbe-d55d0e37cfbd'
+        ]
       - id: task.family
         type: string
         brief: >
-          The task definition family this task definition is a member of.
+          The family name of the [ECS task definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html) used to create the ECS task.
         examples: ['opentelemetry-family']
+      - id: task.id
+        type: string
+        brief: >
+          The ID of a running ECS task, extracted from its ARN.
```

Comment:
Does this need to be extracted from the ARN? If the ARN has this information is there value in pulling it separately in instrumentation?

Reply:
Hi @jsuereth,

This PR relates to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8274 and https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29602. 

In https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8274 I explain that the `aws.ecs.task.id` is already supported / expected by `awsemfexporter`, but the `resourcedetection` didn't populate it. Thus a PR was proposed to fix this withing `resourcedetection` and during the review of that PR, @mx-psi suggested to also mention the new attribute here. So here it is.

In general there are some use cases (like the mentioned `awsemfexporter`) where ECS Task Id is a worthy value, yet AWS ECS Metadata Endpoint does not explicitly advertise it (it does, as you mentioned, advertise the ECS Task ARN, which the ID is a part of), which results in a slightly more complex configuration of `resourcedetectionprocessor` (you have to additionally configure and `extract` action and a regexp parsing the ARN in your OTEL Sidecar config).

Reply:
My point here is that your comment implies it MUST be extracted from the ARN.

What I think you're telling me is it MAY be extracted from the ARN or independently found/used.

I'd like to see the description altered unless you're telling me it is ALWAYS extracted from the ARN.

Reply:
Ahh, okay. Yes. It must be extracted from ARN, there's no other way to obtain it. Task ID is part of the ARN, and is not advertised separately by ECS Metadata Endpoint (which is where all the ECS information in `resourcedetection` is fetched from).

Reply:
That means _today_  `task.id` is conditionally required on the existence of the ARN? I think we need to adapt the requirement levels here to match this.

Also: 

> Ahh, okay. Yes. It must be extracted from ARN, there's no other way to obtain it. Task ID is part of the ARN, and is not advertised separately by ECS Metadata Endpoint (which is where all the ECS information in resourcedetection is fetched from).

Do you think maybe at some point AWS will advertise this task.id via the metadata endpoint you mentioned above? Not sure it makes sense, but we could adapt the description to say something like: At this time, `task.id` must be obtained from `ARN` but later it can be obtained by the metadata endpoint? wdyt @jsuereth

Reply:
@jsuereth I took over the PR for @mkielar . I adapted the description, let me know what you think. I also changed the requirement level to `conditionally_required` on the presence of `task.arn`.

Reply:
I think the way you've modified this makes sense.  There's still something that bothers me about extracting pieces of information from other information and reporting both, but I can see how TaskID isn't the only unique thing in ARN.